db.go

```diff
@@ -64,11 +64,22 @@ func optionsToParams(opts ...map[string]interface{}) (url.Values, error) {
 
 // rowsQuery performs a query that returns a rows iterator.
 func (d *db) rowsQuery(ctx context.Context, path string, opts map[string]interface{}) (driver.Rows, error) {
+	// Queries with large key options can be converted to POST.
+	method := kivik.MethodGet
```

Comment:
I think this new block of code should be moved to after the `options, err := optionsToParams(opts)` line, as `optionsToParams` does some data normalization, which allows for non-string inputs (which is probably a minority use-case, but non-string keys are permitted for views, so we should support them).

Then `options` will be of type `url.Values`.  I'd just check the first value, ignoring multiples (this is what CouchDB does anyway).  So:

    Body: ioutil.NopCloser(strings.NewReader(fmt.Sprintf(`{"keys":%s}`, keys[0])))

or similar.